### ISSUE TITLE
Failed to convert header keys to lower case due to field name conflict: accept-encoding

### ISSUE THREAD

**Issue by oyvindeh** [villadora/express-http-proxy#126]
I get this error when I use express-http-proxy >= 0.9.1:

```
Failed to convert header keys to lower case due to field name conflict: accept-encoding
Error: Failed to convert header keys to lower case due to field name conflict: accept-encoding
    at foo/node_modules/nock/lib/common.js:210:13
    at foo/node_modules/lodash/lodash.js:4332:15
    at baseForOwn (foo/node_modules/lodash/lodash.js:2731:24)
    at Function.forOwn (foo/node_modules/lodash/lodash.js:12042:24)
    at Object.headersFieldNamesToLowerCase (foo/node_modules/nock/lib/common.js:207:5)
    at RequestOverrider (foo/node_modules/nock/lib/request_overrider.js:111:30)
    at new OverriddenClientRequest (foo/node_modules/nock/lib/intercept.js:240:23)
    at foo/node_modules/nock/lib/intercept.js:363:13
    at Object.module.request (foo/node_modules/nock/lib/common.js:137:14)
    at runProxy (foo/node_modules/express-http-proxy/index.js:87:43)
    at invokeCallback (foo/node_modules/raw-body/index.js:262:16)
    at done (foo/node_modules/raw-body/index.js:251:7)
    at IncomingMessage.onEnd (foo/node_modules/raw-body/index.js:307:7)
    at emitNone (events.js:86:13)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:975:12)
```

I see the release notes for 0.9.1 says "Set 'Accept-Encoding' header to match bodyEncoding". Perhaps something goes wrong in relation to this?

**Comment by monkpow** [comment]
Sounds like it.  Can you give me a code snippet that triggers this issue, and I'll fix the bug.  Thanks!

**Comment by btmurrell** [comment]
@monkpow , @oyvindeh  I'm hitting this issue too.  I cannot tell you how to reproduce it -- express (or possibly helmet which I'm using) is loading up my req.headers by some default behavior... however, i do see where the problem in express-http-proxy is happening. 

This may be a suitable fix for express-http-proxy index.js line ~86:
```
      if (bodyEncoding(options)) {
        var acceptEncodingHeader = reqOpt.headers[ 'Accept-Encoding'.toLowerCase() ];
        if (acceptEncodingHeader) {
          reqOpt.headers[ 'Accept-Encoding'.toLowerCase() ] = acceptEncodingHeader + ', ' + bodyEncoding(options);
        } else {
          reqOpt.headers[ 'Accept-Encoding' ] = bodyEncoding(options);
        }
      }
```

**Comment by monkpow** [comment]
@oyvindeh @btmurrell I'm tentative about fixing without being able to see the problem in action.    If your hunch is right, you should be able to skip this chunk of code by adding
```
reqBodyEnconding: null
```
to your options.

**Comment by oyvindeh** [comment]
I'm sorry, I haven't had time to write a snippet that triggers the issue. Hopefully, I'll be able to do so next week.

**Comment by oyvindeh** [comment]
I have problems reproducing this outside of the environment where I get the error (and I cannot share the full code). But I can confirm that 'reqBodyEncoding: null' does indeed work around the issue.

**Comment by monkpow** [comment]
@oyvindeh I did find a relevant bug in the code, which is patch here 861b620fc7476bd6d709630daa1e7872ea7eb6e0.  This fix is in the most recent version (0.10.1) published to npm.    

Let me know if you continue to have issues.  Thanks for the report and clarifications so far.
